errdefs/defs.go

```diff
@@ -52,3 +57,18 @@ type ErrNotImplemented interface {
 type ErrUnknown interface {
 	Unknown()
 }
+
+// ErrCancelled signals that the action was cancelled.
+type ErrCancelled interface {
+	Cancelled()
+}
+
+// ErrDeadline signals that the deadline was reached before the action completed.
+type ErrDeadline interface {
```

Comment:
Would this be a `StatusRequestTimeout`? If so, wondering if it should have a more generic name (but that's a nit)

Reply:
`Request` of what?
This is really for propagating context errors.